src/logging/appInsightsLogger.ts

```diff
@@ -0,0 +1,49 @@
+import { Bag } from "@paperbits/common";
+import { Logger } from "@paperbits/common/logging";
+import { AppInsights } from "applicationinsights-js";
+import { TenantService } from "../services/tenantService";
+
+
+export class LogService implements Logger {
+    constructor(
+        private readonly instrumentationKey: string,
+        private readonly tenantService: TenantService
+    ) {
+        if (this.instrumentationKey && location.hostname.endsWith(".net")) {
+            AppInsights.downloadAndSetup({ instrumentationKey: this.instrumentationKey });
+        }
+        else {
+            console.warn("AppInsights instrumentation key wasn't specified or cannot be used in current environment.");
+        }
+
+        this.traceSession();
+    }
+
+    public async traceSession(): Promise<void> {
+        this.traceEvent(`Session started`);
+    }
+
+    public async traceEvent(eventName: string, properties?: Bag<string>, measurments?: Bag<number>): Promise<void> {
+        AppInsights.trackEvent(eventName, properties, measurments);
+
+        if (!this.instrumentationKey && location.hostname === "localhost") {
```

Comment:
maybe extract this condition **(!this.instrumentationKey && location.hostname === "localhost")** into one small function

Reply:
Removed console logging.